client/js/student-request.js

```diff
@@ -1,24 +1,43 @@
 function getModules(student, id, label) {
   // student does NOT exist, display warning message and cancel request
-  if (student.studentModules === null) {
-    console.log('Student data does not exist.');
+  if (student.studentModules === null || student.studentModules === undefined) {
     label.textContent = `Sorry the student ${id} does not exist`;
     return null;
   }
+  label.textContent = `Module marks for ${id} have been loaded`;
+
+  const LEVEL_5 = [];
 
   const sModules = student.studentModules.modules;
-  // eslint-disable-next-line no-restricted-syntax
-  for (const module in sModules) {
-    console.log(sModules[module]);
+  // confirm all modules are from level 5
+  const is_LVLFive = Object.values(sModules).every(module => module.level === 5);
+  
+  // confirm whether module grades are from level 5 or 6
+  if (is_LVLFive) {
+    for (const module in sModules) {;
```

Comment:
We could replace this `for in` loop to an array iteration method.
This would create cleaner code and reduce the number of loops needed. Improve performance and clean up the code.
The following JS works on my end and achieves this goal:
```js
if (is_LVLFive) {
    const LEVEL_5 = [];
    
    Object.values(sModules).forEach((module, i) => {
      const modMarks = {
        name: module.name,
        marks: module.marks.agreedModuleMark
      };
      LEVEL_5.push(modMarks);
  
      const modNameInput = document.querySelector(`#l5name${i+1}`);
      const moduleMarksInput = document.querySelector(`#l5mark${i+1}output`);
      modNameInput.value = modMarks.name;
      moduleMarksInput.value = modMarks.marks;
    });
    
    console.log(LEVEL_5);
  }
  ```